src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestParser.cs

```diff
@@ -97,15 +99,21 @@ private void ConsumeCurrentTokenInto(HttpSyntaxNode node)
                     ConsumeCurrentTokenInto(node);
                 }
                 else if (CurrentToken is { Kind: HttpTokenKind.Punctuation } and { Text: "#" } &&
-                    !(NextToken is { Kind: HttpTokenKind.Punctuation } and { Text: "#" } &&
-                    NextNextToken is { Kind: HttpTokenKind.Punctuation } and { Text: "#" }))
+                         !(NextToken is { Kind: HttpTokenKind.Punctuation } and { Text: "#" } &&
+                           NextNextToken is { Kind: HttpTokenKind.Punctuation } and { Text: "#" }))
                 {
-                    node.Add(ParseComment());
+                    if (ParseComment() is { } commentNode)
+                    {
+                        node.Add(commentNode);
+                    }
                 }
                 else if (CurrentToken is { Kind: HttpTokenKind.Punctuation } and { Text: "/" } &&
```

Comment:
minor: This `else if` block can be combined with the above one and this would ensure that we only have one call site for `ParseComment`/

Reply:
I considered it. I'm not comfortable with the amount of repetition in the lookahead code and combining it made it harder to read, while `ParseComment` doesn't actually validate that we're in a comment. This factoring looks wrong so I tend to preserve the exploded code paths in order to see clearer branch coverage until that's corrected.

Reply:
The check we perform here will get even more interesting with support for named requests as we will need to reject comments that contain the `# @requestname` syntax :)

Reply:
Yeah, that's another good reason to consolidate the lookaheads into the `ParseComment` (or future `ParseCommentOrNamedRequest`) method.